Marlin/Marlin_main.cpp

```diff
@@ -4137,11 +4260,7 @@ inline void gcode_M42() {
     float  X_current = current_position[X_AXIS],
            Y_current = current_position[Y_AXIS];
 
-    #if ENABLED(Z_PROBE_ALLEN_KEY)
-      const bool stow_probe_after_each = false;
-    #else
-      bool stow_probe_after_each = code_seen('E');
-    #endif
+    bool stow_probe_after_each = code_seen('E');
```

Comment:
This does work now because all the probes are returning (or do not leave) now to their initial xy position (or do not leave).